core/commands/config.go

```diff
@@ -75,21 +75,14 @@ Set the value of the 'datastore.path' key:
 		var output *ConfigField
 		if len(args) == 2 {
 			value := args[1]
-
-			if parseJson, _, _ := req.Option("json").Bool(); parseJson {
-				var jsonVal interface{}
-				if err := json.Unmarshal([]byte(value), &jsonVal); err != nil {
-					err = fmt.Errorf("failed to unmarshal json. %s", err)
-					res.SetError(err, cmds.ErrNormal)
-					return
-				}
-
-				output, err = setConfig(r, key, jsonVal)
-			} else if isbool, _, _ := req.Option("bool").Bool(); isbool {
-				output, err = setConfig(r, key, value == "true")
-			} else {
-				output, err = setConfig(r, key, value)
+			var jsonVal interface{}
+			if err := json.Unmarshal([]byte(value), &jsonVal); err != nil {
+				err = fmt.Errorf("failed to unmarshal json. %s", err)
+				res.SetError(err, cmds.ErrNormal)
+				return
```

Comment:
accept stand-alone strings, numbers (RFC 7159):
- Go 1.3+
- Major browsers (IE 8+, everything else no IE)[1]
- Python 2.7+[1]

else (RFC 4627):
- Ruby 2.1.1 stdlib "json" [2]

Pending:
http://json.org/ (150-ish of those)

[1] http://stackoverflow.com/questions/18202532/can-a-single-stand-alone-literal-form-a-valid-json-document
[2] manual check & "JSON.generate only allows objects or arrays to be converted to JSON syntax." in http://ruby-doc.org/stdlib-2.0.0/libdoc/json/rdoc/JSON.html

---

Or since RFC 7159 supersedes RFC 4627, any legacy system will move toward it anyway.